File: src/LegacyVault.API/Models/Entities/Page.cs
using LegacyVault.API.Models.Enums;

namespace LegacyVault.API.Models.Entities;

public class Page
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid CategoryId { get; set; }
    public PageType Type { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Content { get; set; } = "{}";  // JSON plaintext or Base64 ciphertext
    public bool IsEncrypted { get; set; }
    public string? EncryptionSalt { get; set; }  // Only set for page-level (not category-level) encryption
    public string? EncryptionIV { get; set; }    // Base64-encoded 96-bit AES-GCM IV
    public int SortOrder { get; set; }
    public bool IsFavorite { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public Category Category { get; set; } = null!;
    public ICollection<Attachment> Attachments { get; set; } = new List<Attachment>();
}

File: src/LegacyVault.Android/app/src/main/kotlin/com/legacyvault/app/domain/model/Page.kt
package com.legacyvault.app.domain.model

import com.legacyvault.app.domain.model.enums.PageType

data class PageSummary(
    val id: String,
    val categoryId: String,
    val type: PageType,
    val title: String,
    val isEncrypted: Boolean,
    val isFavorite: Boolean,
    val sortOrder: Int,
    val updatedAt: String
)

/**
 * Full page including content. [content] is raw JSON for unencrypted pages,
 * or Base64 ciphertext for encrypted ones. Decryption happens in the VM layer.
 */
data class Page(
    val id: String,
    val categoryId: String,
    val type: PageType,
    val title: String,
    val content: String,
    val isEncrypted: Boolean,
    val encryptionIV: String?,
    val isFavorite: Boolean,
    val sortOrder: Int,
    val createdAt: String,
    val updatedAt: String
) {
    fun toSummary() = PageSummary(
        id          = id,
        categoryId  = categoryId,
        type        = type,
        title       = title,
        isEncrypted = isEncrypted,
        isFavorite  = isFavorite,
        sortOrder   = sortOrder,
        updatedAt   = updatedAt
    )
}
